### ISSUE TITLE
polyfill.io/v3/polyfill.min.js sometimes returns js code, sometimes doesn't

### ISSUE THREAD

**Issue by saward** [JakeChampion/polyfill-service#2699]
Bug report

## What

Sometimes, polyfill.io/v3/polyfill.min.js returns a js bundle, sometimes just the comment.

In react hook forms, we get an error 'e is not iterable'.

## Details

We had a new error pop up that hadn't happened before.  Sometimes, our site would fail to work, a conflict with react hook forms.  After investigating, it seems to be related to our use of polyfill.io/v3/polyfill.min.js.  If you load that url in your browser directly, and hard-refresh multiple times, you'll see sometimes it returns a file with just the comment, and sometimes the bundle.

However, if we anchor it to a specific version (e.g., https://polyfill.io/v3/polyfill.min.js?version=3.110.1), then it returns the file with just the comment and no js -- which I assume is the desired behaviour for the browser I'm testing from.  Anchoring to a version seems to fix the issue and our site works again.

**Comment by saward** [comment]
I should have added, this behaviour was observed on Chrome+Edge, Firefox, and Safari, across both Windows and Mac operating systems.

**Comment by pureZjr** [comment]
same question

**Comment by loke-dev** [comment]
Experiencing this as well. Like you mentioned, setting the version fixes the issue. Thank you for that tip!

**Comment by nickpolet** [comment]
Experiencing this as well.

**Comment by elias75015** [comment]
Same here, experienced by a lot of users on our national website (geoportail.gouv.fr).

Do you know when this problem could be fix ?
For the moment, we will set the version of the polyfill service as an hotfix as suggested above.

Thanks for your work

**Comment by JakeChampion** [comment]
Hi all,

this looks to be a caching issue in our newer CDN configuration for polyfill.io we are using. I've reverted to the older CDN configuration to bring back the correct behavior whilst we look into fixing this

**Comment by fresh5447** [comment]
@JakeChampion thanks for the update. Do you still recommend locking polyfill version to `https://polyfill.io/v3/polyfill.min.js?version=3.110.1` to avoid further complications? Or do you have confidence you have pinpointed the exact issue?

**Comment by JakeChampion** [comment]
> @JakeChampion thanks for the update. Do you still recommend locking polyfill version to `https://polyfill.io/v3/polyfill.min.js?version=3.110.1` to avoid further complications? Or do you have confidence you have pinpointed the exact issue?

pinning the version shouldn't fix the issue. a way to fix the issue is to add `&use-compute-at-edge-backend=no` to the polyfill.io url as that will force the request to use our old cdn configuration.

we have found the exact issue and are deploying it to our testing environment, once we proved this fixes the issue we will deploy to production and turn on the new CDN config once again.
